File: generators/app/templates/java/app/src/main/java/org/onepointzero/androidstarterjava/support/di/AppModule.java
package <%= packageName %>.support.di;

import android.content.Context;

import <%= packageName %>.application.GetStarterData;
import <%= packageName %>.application.MainPresenter;
import <%= packageName %>.domain.StarterDataRepository;
import <%= packageName %>.infraestructure.DummyStarterDataRepository;
import <%= packageName %>.support.config.Configuration;

import dagger.Module;
import dagger.Provides;

@Module
public class AppModule {

    Context context;
    Configuration configuration;

    public AppModule(Context context, Configuration configuration) {
        this.context = context;
        this.configuration = configuration;
    }

    @Provides
    public MainPresenter mainPresenter(GetStarterData getStarterData) {
        return new MainPresenter(getStarterData);
    }

    @Provides
    public StarterDataRepository starterDataRepository() {
        return new DummyStarterDataRepository(context);
    }

    @Provides
    public GetStarterData getStarterData(StarterDataRepository starterDataRepository) {
        return new GetStarterData(starterDataRepository);
    }
}

File: generators/app/templates/kotlin/app/src/main/java/org/onepointzero/androidstarterkotlin/support/di/AppModule.kt
package <%= packageName %>.support.di

import dagger.Module
import dagger.Provides
import <%= packageName %>.application.GetStarterData
import <%= packageName %>.application.MainPresenter
import <%= packageName %>.domain.StarterDataRepository
import <%= packageName %>.infraestructure.DummyStarterDataRepository
import <%= packageName %>.support.Bootstrap
import javax.inject.Singleton

@Module
@Singleton
class AppModule(val app: Bootstrap) {

    @Provides @Singleton fun provideApp() = app

    @Provides fun mainPresenter(getStarterData: GetStarterData)
            = MainPresenter(getStarterData)

    @Provides fun starterDataRepository(): StarterDataRepository
            = DummyStarterDataRepository(app.applicationContext)

    @Provides fun getStarterData(mainRepository: StarterDataRepository)
            = GetStarterData(mainRepository)
}